velox/dwio/common/CachedBufferedInput.h

```diff
@@ -173,6 +177,17 @@ class CachedBufferedInput : public BufferedInput {
   // concerns.
   void readRegion(const std::vector<CacheRequest*>& requests, bool prefetch);
 
+  // SsdRun only reserves a fixed number of bits for size. loadQuantum larger
+  // than that will result in cache error.
+  void mayCheckLoadQuantum() {
+    if (cache_->ssdCache() != nullptr) {
+      VELOX_CHECK_LE(
+          options_.loadQuantum(),
+          1 << cache::SsdRun::kSizeBits,
+          "Load quantum exceeded SSD cache entry size limit.");
```

Comment:
Can you comment this on the corresponding query config as well? Thanks!

Reply:
Could you point me to the query config? 

Also I'm not sure if it will be relevant, as we'll only enforce this limit when ssd cache is enabled. If ssd cache is disabled, we don't care loadQuantum size at all.